JetUtilities/src/L2Creator.cc

```diff
@@ -431,372 +550,374 @@ bool L2Creator::getL3Rsp() {
+  }
+
+  //
+  // check for large reduced chi2's
+  // above 10 is a plain error; between 5 and 10 is a warning
+  //
+  if (bestRChi2 > 5){
+    if (bestRChi2 > 10)
+      cout<<"\t***ERROR***,";
+    else
+      cout<<"\tWARNING,";
+
+    cout<<" fit for histo "<<gabscor->GetName()
+      <<" has a reduced chi2="<<bestRChi2
+      <<" after "<<fitIter<<" iterations"<<endl;
+  }
 }
 
 //______________________________________________________________________________
 void L2Creator::writeTextFileForCurrentAlgorithm() {
-	TString txtfilename = outputDir + era + "_L2Relative_" + ji->getAlias() + ".txt";
-	ofstream fout(txtfilename);
-	fout.setf(ios::right); 
-
-	unsigned int vector_size = 0;
-    if(l2l3) vector_size = vabscor_eta.size(); //For L2L3 Corrections Together
-    else vector_size = vrelcor_eta.size(); //For L2 & L3 Corrections Separate
-    for (unsigned int ieta=0;ieta<vector_size;ieta++) { 
-    	TGraph* grelcor;
-      	if(l2l3) grelcor = vabscor_eta[ieta]; //For L2L3 Corrections Together
-      	else grelcor = vrelcor_eta[ieta]; //For L2 & L3 Corrections Separate
-      	TF1* frelcor = (TF1*)grelcor->GetListOfFunctions()->Last();
-      	if(frelcor!=0) {
-      		if(ieta==0 || (ieta==1 && delphes))
-               fout<<"{1 JetEta 1 JetPt max(0.0001,"<<frelcor->GetTitle()<<") Correction L2Relative}"<<endl;
-               //fout<<"{1 JetEta 1 JetPt max(0.0001,"<<frelcor->GetExpFormula()<<") Correction L2Relative}"<<endl;
-      		double  etamin  = hl_jetpt.minimum(0,ieta);
-      		double  etamax  = hl_jetpt.maximum(0,ieta);
-      		double  ptmin = grelcor->GetX()[0];
-      		double  ptmax = grelcor->GetX()[grelcor->GetN()-1];
-      		fout<<setw(11)<<etamin
-      		<<setw(11)<<etamax
-        	    <<setw(11)<<(int)(frelcor->GetNpar()+2) //Number of parameters + 2 
-        	    <<setw(12)<<ptmin
-        	    <<setw(12)<<ptmax;
-        	    for(int p=0; p<frelcor->GetNpar(); p++) {
-        	    	fout<<setw(13)<<frelcor->GetParameter(p);
-        	    }
-        	    fout<<endl;
-        }
+  TString txtfilename = outputDir + "/" + era + "_L3Absolute_" + ji->getAlias() + ".txt";
+  ofstream fout(txtfilename);
+  fout.setf(ios::right); 
+
+  unsigned int vector_size = 0;
+  if(l2l3) vector_size = vabscor_eta.size(); //For L2L3 Corrections Together
+  else vector_size = vrelcor_eta.size(); //For L2 & L3 Corrections Separate
+  for (unsigned int ieta=0;ieta<vector_size;ieta++) { 
+    TGraph* grelcor;
+    if(l2l3) grelcor = vabscor_eta[ieta]; //For L2L3 Corrections Together
+    else grelcor = vrelcor_eta[ieta]; //For L2 & L3 Corrections Separate
+    TF1* frelcor = (TF1*)grelcor->GetListOfFunctions()->Last();
+    if(frelcor!=0) {
+      if(ieta==0 || (ieta==1 && delphes)){
+        if(cbPar==17) fout<<"{1 JetEta 1 JetPt max(-1.,"<<frelcor->GetExpFormula()<<") Correction L3Absolute}"<<endl;
+	else          fout<<"{1 JetEta 1 JetPt max(0.0001,"<<frelcor->GetExpFormula()<<") Correction L3Absolute}"<<endl;
```

Comment:
I know it was there before, but we just (3 weeks ago) ran into some problems with GetExpFormula. Apparently something changed in the latest ROOT releases. I'm just trying to save you the headache we ran into.

As for the L2 vs L3 question, see the discussion above.